onebusaway-gtfs-transformer/src/main/java/org/onebusaway/gtfs_transformer/impl/AnomalyCheckFutureTripCounts.java

```diff
@@ -0,0 +1,314 @@
+        }
+        for (Date date : datesToIgnore) {
+            _log.info("ignore: " + date.toString());
+        }
+        for (String key : dayAvgTripsMap.keySet()){
+            System.out.println("key: " + key + " value: " + dayAvgTripsMap.get(key));
+        }
+
+
+        ExternalServices es =  new ExternalServicesBridgeFactory().getExternalServices();
+
+        String[] days = {"Monday","Tuesday","Wednesday","Thursday","Friday","Saturday","Sunday","Holiday"};
+
+
+        Map<Date, Integer> dateTripMap = getDateTripMap(dao);
+
+
+        int dayCounter = 0;
+        Map<String, ArrayList<Double>> dayAvgTripsMapUpdate = new HashMap<String, ArrayList<Double>>();
+
+
+        for (String day:days){
+            dayAvgTripsMapUpdate.put(day, new ArrayList<Double>());
+        }
+
+        while (true) {
+            Date day = removeTime(addDays(new Date(), dayCounter));
+            if (dateTripMap.get(day) == null){
+                break;
+            }
+            int tripCount = dateTripMap.get(day);
+            if ((tripCount < dayAvgTripsMap.get(dayOfWeekFormat.format(day))*(1+percentageMatch/100)) &&
+                    (tripCount > dayAvgTripsMap.get(dayOfWeekFormat.format(day))*(1-percentageMatch/100))){
+                _log.info(day + " has " + tripCount + " trips, and that's within reasonable expections");
+                dayAvgTripsMapUpdate.get(dayOfWeekFormat.format(day)).add((double) tripCount);
+            }
+            else if (holidays.contains(day) & (tripCount < dayAvgTripsMap.get("Holiday")*(1+percentageMatch/100)
+                    && (tripCount > dayAvgTripsMap.get("Holiday")*(1-percentageMatch/100)))) {
+                _log.info(day + " has " + tripCount + " trips, is a holiday, and that's within reasonable expections");
+                dayAvgTripsMapUpdate.get("Holiday").add((double) tripCount);
+            }
+            else if (datesToIgnore.contains(day)) {
+                _log.info(day + " has " + tripCount + " trips, and we are ignoring this possible anomoly");
+            }
+            else {
+                _log.info(day + " has " + tripCount + " trips, this may indicate a problem.");
+                if (!silentMode) {
+                    es.publishMessage(getTopic(), day.toString() + " has " + tripCount + " trips, this may indicate a problem.");
+                }
+            }
+            dayCounter ++;
+        }
+
+
+        for (String day: days) {
+            try {
+                double tripsUpcoming = dayAvgTripsMapUpdate.get(day).stream().mapToDouble(a -> a).average().getAsDouble();
+                double currentAvg = dayAvgTripsMap.get(day);
+                int aprox_hours_per_month = 28 * 24;
+                double suggestedAvg = currentAvg + ((tripsUpcoming - currentAvg) / (aprox_hours_per_month));
+                _log.info(day + "," + suggestedAvg);
+            } catch (Exception e) {
+
```

Comment:
To be clear, this is cheating/lazy programming.  In mission critical top level code, we often catch "Exception" so the program can attempt to recover.  For a specific block, we either shouldn't cause an exception or we should expect a type of exception and deal with it appropriately (aka at least logging the exception!)

If this is happening because of your usage of the stream API, then please expand into procedural code.